common/models/user.js

```diff
@@ -604,8 +604,8 @@ module.exports = function(User) {
     };
 
     UserModel.setter.password = function(plain) {
-      if (typeof plain !== 'string') {
-        return;
+      if (typeof plain === 'number') {
+        plain = plain.toString();
       }
```

Comment:
The enforcement of at least one non-numeric character is a bug as labelled in #2324. The hashing algorithm requires a string of characters but a number should be coerced to a string and that should be the outcome of this PR.

I just want to be clear that the current behavior is not what is expected.

@notbrain, looking forward to seeing the rest of your changes.